Pages/FoodCustomer/Cart.cshtml.cs

```diff
@@ -51,7 +51,8 @@ public async Task<IActionResult> OnPostAddToCart(string id)
             Food food = _context.FoodData.FirstOrDefault(db => db.ID.ToString() == id);
             ShoppingCart result = _context.Cart.FirstOrDefault(c =>
                     c.ApplicationUserId == _userManager.GetUserId(User) &&
-                    c.FoodId == food.ID);
+                    c.FoodId == food.ID &&
+                    c.Reserved == false);
```

Comment:
Maybe this would be easier to read if everything is in one line (unusual to see written like this). What do you think?